mvcc/watcher.go

```diff
@@ -99,6 +100,12 @@ type watchStream struct {
 // Watch creates a new watcher in the stream and returns its WatchID.
 // TODO: return error if ws is closed?
 func (ws *watchStream) Watch(key, end []byte, startRev int64, fcs ...FilterFunc) WatchID {
+	// prevent wrong range where key >= end lexicographically
+	// except special case for watch from key '\x00'
+	if len(end) != 0 && !bytes.Equal(end, []byte{0}) && bytes.Compare(key, end) != -1 {
```

Comment:
`if len(end) != 0 && bytes.Compare(key, end) != -1 &&  !bytes.Equal(end, []byte{0})`? common path is going to be key < end, right?